Daemon crashing during config reload

Noticed some daemon crash during config reload on 4/13.

Error messages (timestamp is UTC)
```
22:56:40.210 daemon.func1 ▶ ERRO 39515 [engine] reverting config due to fatal failure failed to encode for deep copy: gob: type not registered for interface: regexp.Regexp
```

Followed by stacktrace
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7cf6f8]

goroutine 1864758 [running]:
github.com/op/go-logging.(*Logger).IsEnabledFor(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:140
github.com/op/go-logging.(*Logger).log(0x0, 0x1, 0xc000ba33f0, 0xc0000e61a0, 0x2, 0x2)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:144 +0x38
github.com/op/go-logging.(*Logger).Errorf(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:214
github.com/honeydipper/honeydipper/internal/service.(*Service).Reload.func1(0xc0000cb180)
	/build/internal/service/service.go:280 +0x2ac
panic(0xc98c60, 0x130c470)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/op/go-logging.(*Logger).IsEnabledFor(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:140
github.com/op/go-logging.(*Logger).log(0x0, 0x2, 0xc000ba33c0, 0xc000ba33d0, 0x1, 0x1)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:144 +0x38
github.com/op/go-logging.(*Logger).Warningf(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:224
github.com/honeydipper/honeydipper/internal/service.(*Service).loadFeature.func1(0xc0000cb180, 0xc0007232c0, 0x11, 0xc000a4de98)
	/build/internal/service/service.go:142 +0x117
panic(0xc98c60, 0x130c470)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/op/go-logging.(*Logger).IsEnabledFor(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:140
github.com/op/go-logging.(*Logger).log(0x0, 0x2, 0xc000ba3380, 0xc0000e6180, 0x2, 0x2)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:144 +0x38
github.com/op/go-logging.(*Logger).Warningf(...)
	/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/logger.go:224
github.com/honeydipper/honeydipper/internal/service.(*Service).loadFeature(0xc0000cb180, 0xc0007232c0, 0x11, 0xc00009ed00, 0x0, 0x0, 0x0, 0x0)
	/build/internal/service/service.go:160 +0x22d
github.com/honeydipper/honeydipper/internal/service.(*Service).loadRequiredFeatures(0xc0000cb180, 0xc0006ae480, 0x0)
	/build/internal/service/service.go:340 +0xd1
github.com/honeydipper/honeydipper/internal/service.(*Service).Reload(0xc0000cb180)
	/build/internal/service/service.go:287 +0x19d
created by main.reload
	/build/cmd/honeydipper/main.go:137 +0x99
```

--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -213,6 +213,20 @@ func (c *Config) AdvanceStage(service string, stage int, fns ...dipper.ItemProce
 		}
 
 		dipper.Logger.Infof("[%s] service reached stage %s", service, StageNames[stage])
+
+		if c.Stage >= stage {
+			return
+		}
+
+		defer func() {
+			if r := recover(); r != nil {
+				// upon error signal other services to cancel
+				// by setting c.Locker to nil
+				c.Locker = nil
+				panic(r)
+			}
+		}()
+
 		for _, fn := range fns {
 			c.RecursiveStaged(fn)
 		}
@@ -425,16 +439,15 @@ func DeepCopy(s interface{}, d interface{}) error {
 	return nil
 }
 
-func mergeDataSet(d *DataSet, s DataSet) error {
+func mergeDataSet(d *DataSet, source DataSet) error {
+	s := DataSet{}
+	dipper.Must(DeepCopy(source, &s))
 	for name := range s.Systems {
-		system := s.Systems[name]
-		srcCopy := &System{}
-		dipper.Must(DeepCopy(&system, srcCopy))
 		exist, ok := d.Systems[name]
 		if ok {
-			dipper.Must(mergeSystem(&exist, *srcCopy))
+			dipper.Must(mergeSystem(&exist, s.Systems[name]))
 		} else {
-			exist = *srcCopy
+			exist = s.Systems[name]
 		}
 		if d.Systems == nil {
 			d.Systems = map[string]System{}
@@ -442,8 +455,6 @@ func mergeDataSet(d *DataSet, s DataSet) error {
 		d.Systems[name] = exist
 	}
 	s.Systems = map[string]System{}
-	s.Contexts = dipper.MustDeepCopyMap(s.Contexts)
-	s.Drivers = dipper.MustDeepCopyMap(s.Drivers)
 
 	return mergo.Merge(d, s, mergo.WithOverride, mergo.WithAppendSlice)
 }

